Bento/Diff/CollectionViewSectionDiff.swift

```diff
@@ -0,0 +1,108 @@
+import UIKit
+import FlexibleDiff
+
+struct CollectionViewSectionDiff<SectionId: Hashable, RowId: Hashable> {
+    private let oldSections: [Section<SectionId, RowId>]
+    private let newSections: [Section<SectionId, RowId>]
+
+    init(oldSections: [Section<SectionId, RowId>],
+         newSections: [Section<SectionId, RowId>]) {
+        self.oldSections = oldSections
+        self.newSections = newSections
+    }
+
+    func apply(to collectionView: UICollectionView, completion: (() -> Void)? = nil) {
+        let diff = SectionedChangeset(previous: oldSections,
+                                      current: newSections,
+                                      sectionIdentifier: { $0.id },
+                                      areMetadataEqual: Section.hasEqualMetadata,
+                                      items: { $0.rows },
+                                      itemIdentifier: { $0.id },
+                                      areItemsEqual: ==)
+        apply(diff: diff, to: collectionView, completion: completion)
+    }
+
+    private func apply(diff: SectionedChangeset, to collectionView: UICollectionView, completion: (() -> Void)?) {
+        collectionView.performBatchUpdates({
+            for (item, section) in diff.sections.mutations.enumerated() {
```

Comment:
Maybe we should define this block as a closure and then call `collectionView.performBatchUpdates` at the end with it, I think that would improve a bit the readability.